player/mcts/mcts_node.py

```diff
@@ -75,7 +75,7 @@ def build_tree_and_get_move(self, should_continue):
                 return node.action
 
             # Simulate
-            state = self.simulate(state)
+            self.simulate(state)
```

Comment:
I think we should be consistent whether we return state and node or not. So maybe we can ensure whether state and node are correctly updated in the methods and if so do not return them from any method?